### PR TITLE
add torch.bernoulli

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes: https://github.com/Lightning-AI/lightning-thunder/issues/170

### CODE REVIEW

File: thunder/torch/__init__.py [hunk 1]
```diff
@@ -478,12 +478,6 @@ def multinomial(
         generator is None, lambda: f"multinomial does not yet support specifying a generator", NotImplementedError
     )
 
-    utils.check(
```
Comment (kshitij12345) [inline thunder/torch/__init__.py:481]: redundant - above check verifies the same condition.

File: thunder/tests/opinfos.py [hunk 2]
```diff
@@ -5257,6 +5257,51 @@ def torch_randn_like_and_zero(*args, **kwargs):
 tensor_creation_ops.append(randn_like_opinfo)
 
 
+def bernoulli_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make_t = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad, low=0, high=1)
```
Comment (mruberry) [inline thunder/tests/opinfos.py:5261]: Nice generator

File: thunder/tests/opinfos.py [hunk 3]
```diff
@@ -5257,6 +5257,51 @@ def torch_randn_like_and_zero(*args, **kwargs):
...
+
+    shapes = ((), (2, 2), (2, 0, 1), (1, 2, 3))
+
+    for shape in shapes:
+        yield SampleInput(make_t(shape))
+
+
+def bernoulli_error_generator(op, device, **kwargs):
```
Comment (mruberry) [inline thunder/tests/opinfos.py:5269]: Nice error checks

File: thunder/tests/opinfos.py [hunk 4]
```diff
@@ -5257,6 +5257,51 @@ def torch_randn_like_and_zero(*args, **kwargs):
...
+        RuntimeError,
+        err_msg,
+    )
+
+    err_msg = "bernoulli: out is not None which is currently unsupported"
+    yield (SampleInput(torch.ones(3, 3, device=device), out=torch.ones(3, 3, device=device)), RuntimeError, err_msg)
+
+
+# Helper function for `bernoulli` opinfo.
+# It always returns zero tensors, so that the consistency tests and grad tests pass.
```
Comment (mruberry) [inline thunder/tests/opinfos.py:5285]: Nice comment

Comment (mruberry) [reply thunder/tests/opinfos.py:5285]: fyi @t-vi Who's been thinking about reproducible randomness

File: thunder/torch/__init__.py (hunk 1 above)
Comment (mruberry) [reply thunder/torch/__init__.py:481]: Nice fix

File: thunder/torch/__init__.py [hunk 5]
```diff
@@ -572,6 +566,19 @@ def randn_like(
...
+def bernoulli(a: TensorLike, *, generator=None, out=None):
+    # NOTE: Currently, we don't model randomness
+    utils.check(
+        generator is None,
+        lambda: "bernoulli: generator is not None which is currently unsupported",
+        NotImplementedError,
+    )
+    utils.check(out is None, lambda: "bernoulli: out is not None which is currently unsupported", NotImplementedError)
+    utils.check(dtypes.is_float_dtype(a.dtype), lambda: f"bernoulli only supports floating point dtypes, got {a.dtype}")
+    return (uniform(a.shape, device=a.device, dtype=a.dtype) < a).to(a.dtype)
```
Comment (mruberry) [inline thunder/torch/__init__.py:579]: nit: what about `uniform_like` instead of `uniform` here?

Comment (mruberry) [reply thunder/torch/__init__.py:579]: This is a great decomposition of this operation

**Review by mruberry — APPROVED** [review]
This is great!

**Pull request merged**